upcloud/managed_database.go

```diff
@@ -0,0 +1,693 @@
+package upcloud
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// Managed Database states
+const (
```

Comment:
Is there a reason to keep all of these as strings and not as custom enum(ish) types?

Reply:
Doing enums would also make these maybe a little less verbose and hopefully bit easier to understand and use? eg. `StateRunning ManagedDatabaseState = "running"`? (or even just `Running ManagedDatabaseState = "running"`)

Reply:
I would certainly like to do that. However, as everything is in upcloud -package, wouldn't that be too generic name for an exported variable? We will certainly later have other constants that would like to be called `StateRunning`.

Reply:
Ahh, right, yeah, can't really do that then, I was still a bit in -libs mode after reading through it lately..  So, no win for verbosity but types would be cool!

Reply:
string aliases ok or would insist in enums? Those increase boilerplate a bit.

Reply:
I would prefer them to be typed, mostly for DX purposes (eg. discoverability, completion, etc) and to avoid typo-related bugs, missed (manual/custom) strings in (possible future) refactoring, etc.

Reply:
yes but to clarify: you mean an int type with a custom TextMarshaler / TextUnmarshaler?

Reply:
ohhhh right, sorry, misunderstood you, must be the hour... type aliased strings are exactly what I meant, my bad.

Reply:
implemented